### DISCUSSION TITLE
How to use the function to Compress EFI ROM images? / building multi arch iso

### DISCUSSION THREAD

**Discussion by woa7** [ipxe/ipxe#387] [Q&A] [answered]
is the function to  [Compress EFI ROM images](https://github.com/ipxe/ipxe/commit/b76281a8855632df7cc20ad9174e55e29dc6ce67) always used?

i try to build my multi arch iso this way:
```
user@debian:~/ipxe$ make -C src bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn
user@debian:~/ipxe$ $(cd src && ./util/genfsimg -o ipxe-all-arch.iso -s menu.ipxe bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn)
```
or it thear also a simpler way to build a iso with all archs and drivers included?

**Comment by mcb30** [comment, marked as answer]
> is the function to [Compress EFI ROM images](https://github.com/ipxe/ipxe/commit/b76281a8855632df7cc20ad9174e55e29dc6ce67) always used?

It is always used, but only for ROM images.  There is no compression mechanism defined by the UEFI specification for non-ROM images.

> i try to build my multi arch iso this way:
> 
> ```
> user@debian:~/ipxe$ make -C src bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn
> user@debian:~/ipxe$ $(cd src && ./util/genfsimg -o ipxe-all-arch.iso -s menu.ipxe bin-i386-efi/ipxe.efi bin-x86_64-efi/ipxe.efi bin/ipxe.lkrn)
> ```
> 
> or it thear also a simpler way to build a iso with all archs and drivers included?

That is currently the simplest way to do it.

**Reply by woa7** [reply]
thank you
